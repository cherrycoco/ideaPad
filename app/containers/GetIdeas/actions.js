/**
 * Add the idea, this action starts the request saga
 *
 * @return {object} An action object with a type of LOAD_IDEA
 */
export function loadIdeas() {
  return {
    type: 'LOAD_IDEA',
  };
}

/**
 * Dispatched when the idea is successfully added by the request saga
 *
 * @return {object}      An action object with a type of LOAD_IDEA_SUCCESS
 */
export function ideasLoaded(ideas) {
  return {
    type: 'LOAD_IDEA_SUCCESS',
    ideas,
  };
}

/**
 * Dispatched when adding the idea fails
 *
 * @param  {object} error The error
 *
 * @return {object}       An action object with a type of LOAD_IDEA_ERROR passing the error
 */
export function ideasLoadingError(error) {
  return {
    type: 'LOAD_IDEA_ERROR',
    error,
  };
}
