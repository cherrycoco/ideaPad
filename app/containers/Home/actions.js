import { v4 } from 'node-uuid';

/**
 * Add the idea, this action starts the request saga
 *
 * @return {object} An action object with a type of ADD_IDEA
 */
export function addIdea(text, subject) {
  return {
    type: 'ADD_IDEA',
    id: v4(),
    subject,
    text,
  };
}

/**
 * Dispatched when the idea is successfully added by the request saga
 *
 * @return {object}      An action object with a type of ADD_IDEA_SUCCESS
 */
export function ideaAdded() {
  return {
    type: 'ADD_IDEA_SUCCESS',
  };
}

/**
 * Dispatched when adding the idea fails
 *
 * @param  {object} error The error
 *
 * @return {object}       An action object with a type of ADD_IDEA_ERROR passing the error
 */
export function ideaAddingError(error) {
  return {
    type: 'ADD_IDEA_ERROR',
    error,
  };
}
