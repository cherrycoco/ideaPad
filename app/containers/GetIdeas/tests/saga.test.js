/**
 * Tests for loadIdeas sagas
 */

import { put, takeLatest } from 'redux-saga/effects';
import { ideasLoadingError } from '../actions';

import root, { loadIdeas } from '../saga';

const testData = [{
  id: 'test',
  subject: 'test',
  text: 'test',
}];

/* eslint-disable redux-saga/yield-effects */
describe('loadIdeas Saga', () => {
  const loadIdeasGenerator = loadIdeas();

  it('should dispatch the ideasLoaded action if it requests the data successfully', () => {
    expect(loadIdeasGenerator.next().value).toMatchSnapshot();
    expect(loadIdeasGenerator.next(testData).value).toMatchSnapshot();
  });

  it('should call the ideasLoadingError action if the response errors', () => {
    const response = new Error('Some error');
    const putDescriptor = loadIdeasGenerator.throw(response).value;
    expect(putDescriptor).toEqual(put(ideasLoadingError(response)));
  });
});

describe('loadIdeas Saga', () => {
  const rootSaga = root();

  it('should start task to watch for LOAD_IDEA action', () => {
    const takeLatestDescriptor = rootSaga.next().value;
    expect(takeLatestDescriptor).toEqual(takeLatest('LOAD_IDEA', loadIdeas));
  });
});
