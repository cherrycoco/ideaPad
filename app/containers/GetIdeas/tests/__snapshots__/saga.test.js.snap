// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loadIdeas Saga should dispatch the ideasLoaded action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`loadIdeas Saga should dispatch the ideasLoaded action if it requests the data successfully 2`] = `
Object {
  "@@redux-saga/IO": true,
  "PUT": Object {
    "action": Object {
      "ideas": undefined,
      "type": "LOAD_IDEA_SUCCESS",
    },
    "channel": null,
  },
}
`;
